lib/internal/process/execution.js

```diff
@@ -119,7 +121,12 @@ function evalScript(name, body, breakFirstLine, print, shouldLoadESM = false) {
     });
     if (print) {
       const { log } = require('internal/console/global');
-      log(result);
+
+      if (isPromise(result)) {
+        PromisePrototypeThen(result, (resultValue) => log(resultValue));
+      } else {
+        log(result);
+      }
```

Comment:
Here's my suggestion: we're still logging the raw input, but we wait for the process to exit to do so. I see two upsides for doing this:

- We no longer need a special case for promises.
- it will still show the value with which the promise was fulfilled (if applicable).
- it handles never-settling promises just right, and other edge case (e.g. `node -p 'setTimeout(process.exit,100, 0);timers.promises.setTimeout(200)'`)

```suggestion
      process.on('exit', () => {
        log(result);
      });
```

Reply:
The output will be:
![image](https://github.com/nodejs/node/assets/12551007/caea2ff6-20ba-4f33-a121-e8ed7853230b)

I mean, this PR was initiated by the inspiration from another runtime, do we really want to have different behavior with them in this utility?

If we change to:

```js
process.on('exit', async () => {
  log(await result)
})
```

![image](https://github.com/nodejs/node/assets/12551007/c5dda615-f0c9-4151-a1cf-60fb76c9a337)

Works great and also handles `timeouts`.

Reply:
Please consider copy pasting your console output rather than sharing screenshots, as screenshots might be harder to interpret for e.g. folks using screen readers, and it makes it much harder to copy/paste if someone wants to reproduce what you did.



> do we really want to have different behavior with them in this utility?

I think we do.

If we all agree that my suggestion is an improvement, we could start by landing that, and let's start another discussion to try to find consensus – or if you prefer I can open a PR with my suggestion.

Reply:
I don't see the point in waiting for a decision, on the contrary we should make small incremental improvements. I'll open a PR with my suggestion if that's OK.

Reply:
If you remove the Request Change for this PR, I'm okay with you creating another PR.

I push the changes with `process.on`

Reply:
I've opened https://github.com/nodejs/node/pull/52137, I suggest we fast-track it, and you can rebase you PR on top of it and we can see better what diff your PR introduces for the various cases I suggested there.